src/adagrams.js

```diff
@@ -1,8 +1,99 @@
 const Adagrams = {
+  letters : "abcdefghijklmnopqrstuvwxyz".toUpperCase().split(""),
+  letterQuantity : [9, 2, 2, 3, 12, 2, 3, 2, 9, 1, 1, 4, 2, 6, 8, 2, 1, 6, 4, 6, 4, 2, 2, 1, 2, 1],
+  letterValues: {"A": 1, "E": 1, "I" : 1, "O" : 1, "U": 1, "L": 1, "N": 1, "R" : 1, "S" : 1, "T" : 1, "D" : 2, "G" : 2, "B" : 3, "C" : 3, "M" : 3, "P" : 3, "F" : 4, "H" : 4, "V" : 4, "W" : 4, "Y" : 4, "K" : 5, "J" : 8, "X" : 8, "Q" : 10, "Z" : 10},
   drawLetters() {
-    // Implement this method for wave 1
+    let hand = []
+    for (let i = 0; i < this.letters.length; i+=1) {
+      for (let j = 0; j < this.letterQuantity[i]; j+=1) {
+        hand.push(this.letters[i])
+      }
+    }
+    for (let i = hand.length - 1; i > 0; i -= 1) {
+      const j = Math.floor(Math.random() * (i + 1));
+        [hand[i], hand[j]] = [hand[j], hand[i]];
+    }
+
+    return hand.slice(1,11)              
+},
+
+  usesAvailableLetters(input, lettersInHand) {
+    let result = true
+    for (let i =0; i < input.length; i += 1) {
+      if (lettersInHand.includes(input[i]) === true) {
+         let index = lettersInHand.indexOf(input[i]);
+         lettersInHand.splice(index, 1);
+      }
+      else {
+        result = false
+      }
+    }
+    return result
+  },
+
+  scoreWord(word) {
+    let points = 0;
+    let wordArray = word.toUpperCase().split("");
+
+    wordArray.forEach( (letter) => {
+      points += this.letterValues[letter]
+    });
+
+    if (word.length > 6) {
+      points = points + 8;
+    }
+
+    return points
   },
+  
+  highestScoreFrom(words) {
+    let winningWords = []
+    let winningScore = 0
+
+    words.forEach((word) => {
+      let score = this.scoreWord(word)
+      if (score > winningScore) {
+        winningScore = score
+      }
+    });
+
+    words.forEach((word) => {
+      let score = this.scoreWord(word)
+      if (score === winningScore) {
+        winningWords.push(word)
```

Comment:
This algorithm takes three passes through this data:
- Find the high score
- Find everything that's tied for the highest score
- Break ties

This code can be substantially simplified by doing the tie-breaking inline, and that also lets you do it in a single pass. We have an implementation of this for an [old version of the project](https://github.com/AdaGold/js-scrabble/blob/dpr/solution/scrabble.js) if you're interested.